staging/src/k8s.io/kubectl/pkg/config/v1alpha1/types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Preference stores elements of KubeRC configuration file
+type Preference struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// overrides allows changing default flag values of commands.
+	// This is especially useful, when user doesn't want to explicitly
+	// set flags each time.
+	// +listType=atomic
+	Overrides []CommandOverride `json:"overrides"`
+
+	// aliases allows defining command aliases for existing kubectl commands, with optional default flag values.
+	// If the alias name collides with a built-in command, built-in command always takes precedence.
```

Comment:
note in the docs that flag overrides defined in the `overrides` section do *not* apply to aliases for that same kubectl command (and verify there's a test demonstrating that is the case)

Reply:
Added a note in the API docs. Unit test https://github.com/kubernetes/kubernetes/blob/df8a50d20ef596de5f0b491f42c958f6c373a877/staging/src/k8s.io/kubectl/pkg/kuberc/kuberc_test.go#L804-L852 were added previously.